admin/lint_json.m

```diff
@@ -8,27 +8,35 @@ function lint_json(filesin, outputfile)
 % outputfile           char array of filename to write output to (will overwrite)
 %                          if outputfile is empty will write to stdout
 
-    if ~exist('filesin', 'var') || (exist('filesin', 'var') && isempty(filesin))
-        % Default to glob all
-        filesin = {['**', filesep, '*.m']};
-    end
-    if ~exist('outputfile', 'var') || (exist('outputfile', 'var') && isempty(outputfile))
+    p = inputParser;
+    addOptional(p, 'filesin', {['**', filesep, '*.m']}, @iscellstr);
+    addOptional(p, 'outputfile', '_screen', @ischar);
+    addParameter(p, 'exclude', {}, @iscellstr);
+    parse(p, filesin, outputfile, varargin{:});
+
+    if strcmp(p.Results.outputfile, '_screen')
         fh = 1;
     else % Open file
-        fh = fopen(outputfile,'w');
+        fh = fopen(p.Results.outputfile,'w');
         if fh == -1
-            error("MATLAB:FileOpenError", "Failed to open file %s", outputfile);
+            error("MATLAB:FileOpenError", "Failed to open file %s", p.Results.outputfile);
         end
         cleanup = onCleanup(@()(fclose(fh)));
     end
 
     files = [];
-    for i = 1:numel(filesin)
-        flist = dir(filesin{i});
+    for i = 1:numel(p.Results.filesin)
+        flist = dir(p.Results.filesin{i});
```

Comment:
you reassign the `flist` variable multiple times in this loop and change its data type (from filename to filepath). Can you use distinct variables?